File: frontend/src/types/subscription.ts
import type { Plan } from './plan';
import type { User } from './user';

export interface Subscription {
  id: number;
  user_id: number;
  plan_id: number;
  start_date: string;
  end_date: string;
  subscription_url?: string;
  user: User;
  plan: Plan;
}

File: backend/app/models/subscription.py
from sqlalchemy import Column, Integer, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from backend.app.db.base_class import Base
from datetime import datetime, timezone


class Subscription(Base):
    __tablename__ = "subscriptions"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    plan_id = Column(Integer, ForeignKey("plans.id"), nullable=False)

    start_date = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))
    end_date = Column(DateTime(timezone=True), nullable=False)

    user = relationship("User", back_populates="subscriptions")
    plan = relationship("Plan")